### PR TITLE
Required fields on register, login and comments forms

### LINKED ISSUE #6078: Validate non-empty comment client side
Watch for things like `<p>&nbsp;</p>`

### PR DESCRIPTION
**Pull request by jdalsem**
partially fixes #6078

### CODE REVIEW

**Comment by mrclay** [comment]
Is a password field ever optional?

**Comment by jdalsem** [comment]
> Is a password field ever optional?

yeah sure why not? Maybe a settings form which stores credentials? Or update your account data in Elgg (user) settings. You are allowed to leave it blank there...

**Comment by ewinslow** [comment]
Yes, on the install flow it is optional to make dev easier.

For user logins I think no because there is a minimum length requirement, but that should only be enforced when creating the password, not when typing it in. In case the requirement changes to become more strict and then the user can't type their password anymore!

File: views/default/forms/comment/save.php [hunk 1]
```diff
@@ -59,6 +59,7 @@
 	$comment_input = elgg_view('input/text', array(
 		'name' => 'generic_comment',
 		'value' => $comment_text,
+		'required' => true
```
Comment (mrclay) [inline views/default/forms/comment/save.php]: I'd much prefer trailing commas after all these, but LGTM.

**Comment by mrclay** [comment]
Couldn't this go in 1.11? or at least 1.x.

**Comment by ewinslow** [comment]
I'm personally fine with it in either 1.x or master.

**Comment by mrclay** [comment]
Agreed. Add trailing commas, or just merge as is.

**Comment by ewinslow** [comment]
Report a bug to add trailing commas for multiline arrays to the linter?

**Pull request merged**